src/unit_tests/variant/variant_cast_test.cpp

```diff
@@ -0,0 +1,241 @@
+*   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE     *
+*   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER          *
+*   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,   *
+*   OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE   *
+*   SOFTWARE.                                                                       *
+*                                                                                   *
+*************************************************************************************/
+
+#include <catch/catch.hpp>
+#include <iostream>
+#include <rttr/type>
+
+using namespace rttr;
+
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+TEST_CASE("variant - variant_cast<T>(variant*)", "[variant]")
+{
+    SECTION("empty cast")
+    {
+        variant var;
+
+        auto value = variant_cast<int>(&var);
+
+        CHECK(value == nullptr);
+    }
+
+    SECTION("valid cast")
+    {
+        variant var = 12;
+
+        auto value = variant_cast<int>(&var);
+
+        CHECK(value != nullptr);
+
+        static_assert(std::is_same<decltype(value), int*>::value, "Must be integer pointer type!");
+
+        REQUIRE(value != nullptr);
+        CHECK(*value == 12);
+    }
+
+    SECTION("valid cast - pointer")
+    {
+        int obj = 12;
+        variant var = &obj;
+
+        auto value = variant_cast<int*>(&var);
+        REQUIRE(value != nullptr);
+        CHECK(**value == 12);
+    }
+
+    SECTION("invalid cast")
+    {
+        variant var = 12;
+
+        auto value = variant_cast<bool>(&var);
+        CHECK(value == nullptr); // nullptr, because not of type "int"
+    }
+}
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+TEST_CASE("variant - variant_cast<T>(const variant*)", "[variant]")
+{
+    SECTION("empty cast")
+    {
+        const variant var;
+
+        auto value = variant_cast<int>(&var);
+
+        CHECK(value == nullptr);
+    }
+
+    SECTION("valid cast")
+    {
+        const variant var = 12;
+
+        auto value = variant_cast<int>(&var);
+
+        CHECK(value != nullptr);
+
+        static_assert(std::is_same<decltype(value), const int*>::value, "Must be integer pointer type!");
+
+        REQUIRE(value != nullptr);
+        CHECK(*value == 12);
+    }
+
+    SECTION("invalid cast")
+    {
+        const variant var = 12;
+
+        auto value = variant_cast<bool>(&var);
+        CHECK(value == nullptr); // nullptr, because not of type "int"
+    }
+}
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+TEST_CASE("variant - variant_cast<T>(variant&)", "[variant]")
+{
+    SECTION("by value")
+    {
+        variant var = 12;
+
+        REQUIRE(var.get_type() == type::get<int>());
+
+        auto value = variant_cast<int>(var);
```

Comment:
;) -> `auto` has the semantic of always stripping `references` == `&`. So I think when you write `auto value` this is always an lvalue (not an lvalue reference). 
So it seems that this test is obsolete :-), but better leave it included :-)

Reply:
I know, but also this case can happen, thats why I added the test.
Its also in the section: `SECTION("by value")`